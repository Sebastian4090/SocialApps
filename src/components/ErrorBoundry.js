import React, { Component } from 'react';

class ErrorBoundry extends Component {
    constructor(){
        super()
        this.state = {
            hasError: false
        }
    }

    componentDidCatch(){
        this.setState({ hasError: true })
    }

    render(){
        if (this.state.hasError){
            return <h1>An Error Has Occurred</h1>
        }
        return this.props.children
    }
}

export default ErrorBoundry;