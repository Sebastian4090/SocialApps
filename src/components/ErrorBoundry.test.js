import { shallow } from 'enzyme';
import React from 'react';
import ErrorBoundry from './ErrorBoundry';

it('expect to render ErrorBoundry component', () => {
    expect(shallow(<ErrorBoundry />)).toMatchSnapshot()
})

it('expect to render ErrorBoundry error', () => {
    const Something = () => {
        return null;
    }

    const wrapper = shallow(<ErrorBoundry ><Something /></ErrorBoundry>)
    const error = new Error('error!')
    wrapper.find(Something).simulateError(error)
    expect(wrapper.state()).toEqual({ hasError: true})
})
