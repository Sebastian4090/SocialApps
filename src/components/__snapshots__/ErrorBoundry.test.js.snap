// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ErrorBoundry component 1`] = `""`;
